Create contribution.md (#42)
rsksmart/devportal: contribution.md

## Open Source Contribution Guide

> Welcome to our open source project! We're excited to have you contribute. This guide will help you understand our contribution process and ensure your contributions are well-received.

Reviewer: ```suggestion
Welcome to Rootstock! 

We're excited to have you contribute to Rootstock Open Source Projects. This guide will help you understand the contribution process and ensure contributors have clear guidelines and procedures for how to contribute.
```

All contributions are welcome, including issues, updates and tweaks, tutorials and more.

### Table of Contents
1. [Code of Conduct](#code-of-conduct)
2. [How to Contribute](#how-to-contribute)
    - [Reporting Bugs](#reporting-bugs)
    - [Suggesting Features](#suggesting-features)
    - [Contributing Code](#contributing-code)
        - [Development Environment Setup](#development-environment-setup)
        - [Coding Standards](#coding-standards)
        - [Commit Messages](#commit-messages)
        - [Submitting Pull Requests](#submitting-pull-requests)
3. [Review Process](#review-process)
4. [Community and Communication](#community-and-communication)
5. [License](#license)

### Code of Conduct

Please read our [Code of Conduct]() before contributing. We are committed to fostering a welcoming and inclusive community.

### How to Contribute

#### Reporting Bugs

> 1. **Check for Existing Issues**: Before reporting a bug, please check the [issue tracker](issues) to see if it has already been reported.

Reviewer: ```suggestion
1. **Check for Existing Issues**: Before reporting a bug, please check the [issue tracker](https://github.com/rsksmart/rsksmart.github.io/issues) to see if it has already been reported.
```

2. **Create a New Issue**: If your issue is not listed, create a new one. Provide as much detail as possible:
    - A clear and descriptive title.
    - Steps to reproduce the bug.
    - Expected and actual results.
    - Any relevant screenshots or logs.
    - Your environment details (OS, browser, version, etc.).

#### Suggesting Features

1. **Check for Existing Requests**: Look through the [issue tracker](issues) to see if someone else has suggested the feature.
2. **Create a New Feature Request**: If the feature is not listed, open a new issue. Include:
    - A clear and descriptive title.
    - The motivation for the feature.
    - Detailed description of the proposed feature.
    - Any relevant mockups or screenshots.

> #### Contributing documentation

Reviewer: ```suggestion
#### Contributing
```

1. **Fork the Repository**: Fork our repo to your GitHub account.
2. **Clone Your Fork**: Clone your fork to your local machine.
    ```bash

>     git clone https://github.com/zgraya-digital/devportal-rootstock.git

Reviewer: ```suggestion
    git clone https://github.com/rsksmart/devportal.git
```

* Limit the first line to 72 characters or less  
* Consider starting the commit message with an applicable emoji:  
  * 🎨 :art: when improving the format/structure of the code  
  * 🐎 :racehorse: when improving performance  
  * 🚱 :non-potable\_water: when plugging memory leaks  
  * 📝 :memo: when writing docs  
  * 🐧 :penguin: when fixing something on Linux  
  * 🍎 :apple: when fixing something on macOS  
  * 🏁 :checkered\_flag: when fixing something on Windows  
  * 🐛 :bug: when fixing a bug  
  * 🔥 :fire: when removing code or files  
  * 💚 :green\_heart: when fixing the CI build  
  * ✅ :white\_check\_mark: when adding tests  
  * 🔒 :lock: when dealing with security  
  * ⬆️ :arrow\_up: when upgrading dependencies  
  * ⬇️ :arrow\_down: when downgrading dependencies  
  * 👕 :shirt: when removing linter warnings

### Review Process

1. **Code Review**: Your PR will be reviewed by our maintainers. They may ask for changes or provide feedback.
2. **Make Necessary Changes**: Address any feedback and update your PR.
3. **Merge**: Once approved, your PR will be merged into the main branch.

### Community and Communication

> 1. **Join Our Discussion**: Participate in discussions and If you need help, reach out to us on [discord](link).

Reviewer: ```suggestion
1. **Join the Discussion**: Participate in discussions and If you need help, reach out to us on [discord](http://discord.gg/rootstock).
```

2. **Stay Updated**: Follow our updates on [social media](https://x.com/rootstocklabs).

### License

By contributing, you agree that your contributions will be licensed under the [MIT License](LICENSE).

---

> Thank you for contributing! Together, we can make this project better for everyone.

Reviewer: ```suggestion
Thank you for contributing to Rootstock!
```
